src/CXXTests/testSidreStorage.cc

```diff
@@ -0,0 +1,645 @@
+    int n = 10;
+    T* rawSidreData;
+
+    Spheral::Field<Spheral::Dim<1>, std::vector<T>> makeField()
+    {
+      Spheral::NodeList<Spheral::Dim<1>> makeNodeList("test bed", n, 0);
+      Spheral::Field<Spheral::Dim<1>, std::vector<T>> testField("test field", makeNodeList);
+      for (int i = 0; i < n; i++)
+        for (int j = 0; j < n; j++)
+          testField[i].emplace_back(i);
+      return testField;
+    }
+
+    void allocRawSidreData(const Spheral::Field<Spheral::Dim<1>, std::vector<T>>& testField)
+    {
+      rawSidreData = myData.alloc_view("SidreTest", testField)->getData();
+    }
+};
+
+TYPED_TEST_SUITE(SidreDataCollectionTestVector, MyTypes);
+
+TYPED_TEST(SidreDataCollectionTestVector, vector)
+{
+  auto testField = this->makeField();
+
+  // for (int i = 0; i < n; i++)
+  // {
+  //   for (int j = 0; j < n; j++)
+  //     std::cout << testField[i][j] << " ";
+  //   std::cout << std::endl;
+  // }
+
+  this->allocRawSidreData(testField);
+
+  //this->myData.printDataStore();
+
+  for (int i = 0; i < this->n; i++)
+    EXPECT_EQ(testField[0][i], this->rawSidreData[i]);
+}
+
+
+
+template <typename T>
+class SidreDataCollectionTestTupleThree : public ::testing::Test
+{
+  public:
+    Spheral::SidreDataCollection myData;
+    int n = 5;
+    T* rawSidreData;
+
+    Spheral::Field<Spheral::Dim<1>, std::tuple<T, T, T>> makeField()
+    {
+      Spheral::NodeList<Spheral::Dim<1>> makeNodeList("test bed", n, 0);
+      Spheral::Field<Spheral::Dim<1>, std::tuple<T, T, T>> testField("test field", makeNodeList);
+      for (int i = 0; i < n; i++)
+        testField[i] = std::make_tuple(i, i, i);
+      return testField;
+    }
+
+    void allocRawSidreData(const Spheral::Field<Spheral::Dim<1>, std::tuple<T, T, T>>& testField)
+    {
+      rawSidreData = myData.alloc_view("SidreTest", testField)->getData();
+    }
+};
+
+TYPED_TEST_SUITE(SidreDataCollectionTestTupleThree, MyTypes);
+
+TYPED_TEST(SidreDataCollectionTestTupleThree, tuple)
+{
+  auto testField = this->makeField();
+
+  // for (int i = 0; i < this->n; i++)
+  //   std::cout << testField[i] << " ";
+  // std::cout << std::endl;
+
+  this->allocRawSidreData(testField);
+
+  // this->myData.printDataStore();
+
+  EXPECT_EQ(std::get<0>(testField[0]), this->rawSidreData[0]);
+  EXPECT_EQ(std::get<1>(testField[0]), this->rawSidreData[1]);
+  EXPECT_EQ(std::get<2>(testField[0]), this->rawSidreData[2]);
+}
+
+
+
+TEST(SidreDataCollectionTestDim1Vector, Dim1Vector)
+{
+  Spheral::SidreDataCollection myData;
+  int n = 10;
+
+  Spheral::NodeList<Spheral::Dim<1>> makeNodeList("test bed", n, 0);
+  Spheral::Field<Spheral::Dim<1>, Spheral::Dim<1>::Vector> testField("test field", makeNodeList);
+  for (int i = 0; i < n; i++)
+    testField[i] = Spheral::Dim<1>::Vector(i);
+  
+  double* rawSidreData = myData.alloc_view("SidreTest", testField)->getData();
+
+  //myData.printDataStore();
+
+  //for (int i = 0; i < n; i++)
+    EXPECT_EQ(testField[0].x(), rawSidreData[0]);
```

Comment:
We should ensure we are checking every element of the sidre data for each of these tests.

Reply:
I do intend to do this, I just don't know the best way to do this right now as these functions aren't used anywhere. Once I start working on the restart I'll know what the functions should return and I can adjust the tests then.

Reply:
I think you could iterate over the number of elements and just ensure that all the data in rawSidreData is as expected. In this case you should be able to do something like:
```
for (int i = 0; i < n; i++)
    EXPECT_EQ(i, rawSidreData[i])
```
for the 3D Vector below you could do:
```
for (int i = 0; i < n; i++)
    EXPECT_EQ(i, rawSidreData[i*3])
    EXPECT_EQ(i+1, rawSidreData[i*3+1])
    EXPECT_EQ(i+2, rawSidreData[i*3+2])
```

Reply:
The way that I am storing the dim types in sidre now is data type per view (a field of dim<1>vectors has multiple views for each vector value) so, rawSidreData only has the first value. I did this because it was the easiest to make it work at the time, but it will change once I start using the functions.